### PR TITLE
HHH-16182 - Converted boolean values not always properly handled in predicates

### PR DESCRIPTION
**Pull request by sebersole**
HHH-16182 - Converted boolean values not always properly handled in predicates

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java [hunk 1]
```diff
@@ -7105,20 +7100,21 @@ private JdbcMappingContainer getBooleanType() {
 	@Override
 	public Object visitBooleanExpressionPredicate(SqmBooleanExpressionPredicate predicate) {
 		final Expression booleanExpression = (Expression) predicate.getBooleanExpression().accept( this );
-		if ( booleanExpression instanceof SelfRenderingExpression ) {
-			final Predicate sqlPredicate = new SelfRenderingPredicate( (SelfRenderingExpression) booleanExpression );
-			if ( predicate.isNegated() ) {
-				return new NegatedPredicate( sqlPredicate );
-			}
-			return sqlPredicate;
-		}
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:7114]: I think this code here was to support functions as predicates, though it seems we lack a test for this. Can you please re-add it or do you think the code path would never be hit?

**Review by beikov — APPROVED** [review]

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java (hunk 1 above)
Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:7114]: I mean, there should be a test...

Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:7108]: Did not hear a reply, so added 2 basic tests of boolean functions wrt predicates -

1. as a predicate - `... where boolean_function`
2. in a predicate - `... where boolean_function = true`

Both work with the code as-is

**Comment by sebersole** [comment]
Applied upstream

**Pull request closed without merging**